### DISCUSSION TITLE
Wavesurfer Events affecting scrollIntoView

### DISCUSSION THREAD

**Discussion by pokemon963** [katspaugh/wavesurfer.js#2729] [Q&A]
Hi! I have been using wavesurfer in correspondence with another API. I have an event handler for when the audio is playing (i.e. `audioprocess`). When the audio the playing, I wish for a certain item in my page from another API to scroll into view. However, the item only gets centered into the screen when the audio is paused. I am not sure why this is happening. I thought it was maybe that the wavesurfer events are synchronous and only activate the scrollIntoView once the event is done. Here is the code I have:

```
wavesurfer.on('audioprocess', function() { // runs continuously when audio plays
        var currentTime = wavesurfer.getCurrentTime();
        var audioTime = (Math.round(currentTime * 1000)/1000);

        document.getElementById('time-current').innerText = audioTime;
        interactWaveform(audioTime);

        if (!osmd.EngravingRules.RenderSingleHorizontalStaffline) {
          var diff = osmd.cursor.cursorElement.getBoundingClientRect().top;
          osmd.cursor.cursorElement.scrollIntoView({behavior: diff < 1000 ? "smooth" : "auto", block: "center"});
        } else {
          osmd.cursor.cursorElement.scrollIntoView({behavior: "smooth", inline: "center"});
        }
});
```

**Comment by katspaugh** [comment]
Audioprocess is a very high-frequency event, esp. if you’re using the default WebAudio playback. So it will fire that event far more many times than you probably expect.

Perhaps try using `backend: ‘MediaElement’` or try this branch where I replaced Web Audio audioprocess with a more render-friendly event: https://github.com/wavesurfer-js/wavesurfer.js/pull/2706

**Comment by pokemon963** [comment]
I tried the first solution with the backend and that did not work. As for the second solution, I am struggling to navigate and find the Web Audio audioprocess you are talking about. Can you provide some more assistance for this please?

**Reply by katspaugh** [reply]
As I said, this even is firing at quite a high frequency (> 60 FPS), so calling `.scrollIntoView` on each event is most certainly not what you want.
